src/main/java/com/google/api/codegen/util/py/PythonNameFormatter.java

```diff
@@ -0,0 +1,93 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.util.py;
+
+import com.google.api.codegen.util.Name;
+import com.google.api.codegen.util.NameFormatter;
+import com.google.api.codegen.util.NamePath;
+
+/** The NameFormatter for Python. */
+public class PythonNameFormatter implements NameFormatter {
+
+  @Override
+  public String publicClassName(Name name) {
+    return name.toUpperCamel();
+  }
+
+  @Override
+  public String privateClassName(Name name) {
+    return publicClassName(name);
+  }
+
+  @Override
+  public String localVarName(Name name) {
```

Comment:
I could also see having each language fill in a table with the functions that implement each of these methods in that language, and then these methods simply call the function specified in the table. That would make the code more declarative and thus more compact and easier to parse.